<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>11_Promise对象</title>
</head>
<body>
<div id="news_wrap"></div>
<div id="comments_wrap"></div>
<script type="text/javascript">

  /**
   * 发送ajax请求
   * @param method 请求方式
   * @param url 请求地址
   * @return {Promise<any>}
   */
  function sendAjax(method, url) {
    return new Promise((resolve, reject) => {
      const xhr = new XMLHttpRequest();
      xhr.onreadystatechange = () => {
        if (xhr.readyState === 4) {
          if (xhr.status === 200) {
            const data = JSON.parse(xhr.responseText);
            //将promise对象状态改为成功的状态
            resolve(data);
          } else {
            //请求失败
            console.log(xhr.responseText);
            //将promise对象状态改为失败的状态
            reject(xhr.responseText);
          }
        }
      }
      xhr.open(method, url);
      xhr.send();
    })
  }

  //发送请求请求新闻数据
  sendAjax('GET', 'http://localhost:3000/news?id=1')
    .then(res => {  //成功的回调
      console.log(res);
      //显示新闻内容
      const newsWrap = document.getElementById('news_wrap');
      newsWrap.innerHTML = `<h1>${res.title}</h1><p>${res.content}</p>`;
      //发送请求请求评论的内容
      /*
      sendAjax('GET', `http://localhost:3000${res.commentsUrl}`)
        .then(res => {
          console.log(res);
          //显示评论的内容
          const commentsWrap = document.getElementById('comments_wrap');
          let html = '';
          res.forEach(item => {
            html += `<p>${item.content}</p>`;
          })
          commentsWrap.innerHTML = html;
        })
        .catch(err => console.log(err))
        */
      return sendAjax('GET', `http://localhost:3000${res.commentsUrl}`);
    })
    .catch(err => {  //失败的回调
      console.log(err);
      return Promise.reject(err);
    })
    .then(res => {
      console.log(res);
      //显示评论的内容
      const commentsWrap = document.getElementById('comments_wrap');
      let html = '';
      res.forEach(item => {
        html += `<p>${item.content}</p>`;
      })
      commentsWrap.innerHTML = html;
    })
    .catch(err => console.log(err))


</script>

</body>

</html>